src/Microsoft.Sbom.Api/Utils/Comparer/SbomEqualityComparer.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Sbom.Common.Spdx30Entities;
+using Microsoft.Sbom.Common.Spdx30Entities.Enums;
+using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Parsers.Spdx22SbomParser.Entities;
+using Microsoft.Sbom.Parsers.Spdx22SbomParser.Entities.Enums;
+using Microsoft.Sbom.Utils;
+using SbomEntities = Microsoft.Sbom.Extensions.Entities;
+
+namespace Microsoft.Sbom.Api.Utils.Comparer;
+
+/// <summary>
+/// Finds the content differences between an SPDX 2.2 document and an SPDX 3.0 document.
+/// </summary>
+public class SbomEqualityComparer
+{
+    private JsonSerializerOptions serializerOptions = new JsonSerializerOptions
+    {
+        Converters = { new ElementSerializer() },
+    };
+
+    private readonly JsonElement spdx22Json;
+    private readonly JsonElement spdx30Json;
+
+    private readonly SbomFileComparer sbomFileComparer;
+    private readonly SbomPackageComparer sbomPackageComparer;
+    private readonly SbomRelationshipComparer sbomRelationshipComparer;
+    private readonly SbomReferenceComparer sbomReferenceComparer;
+
+    public SbomEqualityComparer(JsonElement spdx22Json, JsonElement spdx30Json)
+    {
+        this.spdx22Json = spdx22Json;
+        this.spdx30Json = spdx30Json;
+        this.sbomFileComparer = new SbomFileComparer();
+        this.sbomPackageComparer = new SbomPackageComparer();
+        this.sbomRelationshipComparer = new SbomRelationshipComparer();
+        this.sbomReferenceComparer = new SbomReferenceComparer();
+    }
+
+    public bool DocumentsEqual()
```

Comment:
Is there a cheap way that we can we capture how the documents are different? Maybe a parameter we could use to capture the differences or a parameter that we populate while performing the comparison?

Reply:
How about logging by default whenever there is a difference? I was thinking it would be sufficient to log which part of the SBOM is different. For example, if the files differ we could log that the files don't match.

I'm hesitant to log the details of which files could be different because that could result in a lot of log output since the SBOMs can get pretty large.